sdk/core/ci.yml

```diff
@@ -35,3 +35,4 @@ stages:
       Artifacts:
         - Name: azure-core
           Path: azure-core
+          VcpkgPortName: azure-core-cpp
```

Comment:
Is there a reason for the "-cpp" here? Does this essentially become the package name? Will we also need to update the package index data (https://github.com/Azure/azure-sdk/blob/master/_data/releases/latest/cpp-packages.csv) with this information? At a minimal we will want to update that to point to vcpkg repo now.

Reply:
See https://github.com/Azure/azure-sdk-for-cpp/pull/1013#discussion_r528051335
Its to distinguish between C ports of packages (if we were to add them in the future) and C++ ports.

We want the directory and "package name" on vcpkg to be `azure-core-cpp`, rather than just `azure-core`. That said, we should make sure to flesh out the ramifications of that decision, end to end, and you pose some good questions. We still use `azure-core` as the project name within the repo, and I believe we want the user-experience to be that they use `azure-core` or `azure-storage` in their `CMakeList.txt` files.

@RickWinter and @antkmsft might be able to provide more insights, along with @vhvb1989. From a customer's perspective during package acquisition and reading docs, when would they see and write `azure-core-cpp` and when would they see or write `azure-sdk`. When users want to install the package they would run `vcpkg install <package name>`, i.e. `vcpkg install azure-storage-blobs-cpp`.